implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -139,7 +138,11 @@ private JwtContext parseClaims(String token, JWTAuthContextInfo authContextInfo,
                 builder.setExpectedIssuer(authContextInfo.getIssuedBy());
             }
 
-            if (authContextInfo.getExpGracePeriodSecs() > 0) {
+            // mp.jwt.verify.clock.skew property has precedence over smallrye.jwt.expiration.grace
+            // smallrye.jwt.expiration.grace has default 60 seconds which is used if none of the properties are present
+            if (authContextInfo.getClockSkew() != null && authContextInfo.getClockSkew() > 0) {
+                builder.setAllowedClockSkewInSeconds(authContextInfo.getClockSkew().intValue());
+            } else if (authContextInfo.getExpGracePeriodSecs() > 0) {
```

Comment:
@sberyozkin Let me know if you agree with usage of _Long_ for this property and if you agree with clockSkew having precedence and _expiration.grace_ being ignored if _clock.skew_ is present. Thank you!

Reply:
Hi @Skyllarr Thanks for the PR, should we have this property set to `60` by default ?

Reply:
@sberyozkin I did not do it at fist since I have seen [this older commit](https://github.com/eclipse/microprofile-jwt-auth/pull/286/files) , but yes given the situation changed since it sounds good. I have updated the PR accordingly